src/main/java/edu/tamu/app/service/manager/GitHubMilestoneService.java

```diff
@@ -0,0 +1,115 @@
+import org.kohsuke.github.GHMilestoneState;
+import org.kohsuke.github.GHOrganization;
+import org.kohsuke.github.GHProject;
+import org.kohsuke.github.GHProject.ProjectStateFilter;
+import org.kohsuke.github.GHProjectCard;
+import org.kohsuke.github.GHProjectColumn;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GHUser;
+
+import edu.tamu.app.cache.model.Card;
+import edu.tamu.app.cache.model.Member;
+import edu.tamu.app.cache.model.Sprint;
+import edu.tamu.app.model.ManagementService;
+
+public class GitHubMilestoneService extends AbstractGitHubService {
+
+    public GitHubMilestoneService(final ManagementService managementService) throws IOException {
+        super(managementService);
+    }
+
+    @Override
+    public List<Sprint> getActiveSprintsByScopeId(final String scopeId) throws Exception {
+        logger.info("Fetching active sprints for remote project with scope id " + scopeId);
+        List<Sprint> activeSprints = new ArrayList<Sprint>();
+        GHRepository repo = github.getRepositoryById(scopeId);
+        List<GHProject> projects = repo.listProjects(ProjectStateFilter.OPEN).asList();
+        for (GHProject project : projects) {
+            String sprintId = String.valueOf(project.getId());
+            String projectName = repo.getName();
+            Map<String, List<Card>> partitionedCards = getCards(project);
+            for (Entry<String, List<Card>> partition : partitionedCards.entrySet()) {
+                activeSprints.add(new Sprint(sprintId, partition.getKey(), projectName, partition.getValue()));
+            }
+        }
+        return activeSprints;
+    }
+
+    @Override
+    public List<Sprint> getAdditionalActiveSprints() throws Exception {
+        GHOrganization organization = github.getOrganization(ORGANIZATION);
+        List<GHProject> projects = organization.listProjects(ProjectStateFilter.OPEN).asList();
+        List<Sprint> sprints = new ArrayList<Sprint>();
+        for (GHProject project : projects) {
+            String sprintId = String.valueOf(project.getId());
+            Map<String, List<Card>> partitionedCards = getCards(project);
+            String productName = String.format("%s - %s", organization.getName(), project.getName());
+            int count = 0;
+            for (Entry<String, List<Card>> partition : partitionedCards.entrySet()) {
+                sprints.add(new Sprint(sprintId + "-" + count, partition.getKey(), productName, partition.getValue()));
+                count++;
+            }
+        }
+        return sprints;
+    }
+
+    private Map<String, List<Card>> getCards(GHProject project) throws IOException {
+        Map<String, List<Card>> cardsByMilestone = new HashMap<>();
+        for (GHProjectColumn column : project.listColumns().asList()) {
+            List<GHProjectCard> projectCards = column.listCards().asList();
+            Map<Long, GHIssue> cardContents = new HashMap<>();
+            for (GHProjectCard card : projectCards) {
```

Comment:
I do not see the need for iterating over the project column cards, `projectCards`, twice. Should be able to collect to a map.

I would think something like this would have the same result:
```
.collect(Collectors.toMap(c -> c.getMilestone(), c -> c))
```

Key mapper being the card milestone and the value mapper being the card itself.

Reply:
The getContent method makes a call to the GItHub api. To do that in the stream would require a try/catch block in the lambda to handle the IOException. It seemed easier to read this way.

Reply:
I see. I guess we have to question if we want to fail the entire getCards method if one getContent fails to return. It may make since to create an function to get content that returns an Optional<GHIssue> and try catches the IOException.

Reply:
Yeah, a lot of the methods do this. The other 3 comments I haven't resolved all have the same problem with other methods (listColumns, listProjects, listCards). 

I was looking into using a functional interface to wrap the offending calls, but it's looking like it's not going to work for all  the cases I have.